src/custom/hooks/useSwapCallback.ts

```diff
@@ -2,10 +2,121 @@ import { SwapCallbackState } from '@src/hooks/useSwapCallback'
 import { INITIAL_ALLOWED_SLIPPAGE } from 'constants/index'
 
 // import { useSwapCallback as useSwapCallbackUniswap } from '@src/hooks/useSwapCallback'
-import { Percent, Trade } from '@uniswap/sdk'
+import { ChainId, Percent, Trade, TradeType } from '@uniswap/sdk'
 import { useActiveWeb3React } from '@src/hooks'
 import useENS from '@src/hooks/useENS'
 import { useMemo } from 'react'
+import useTransactionDeadline from '@src/hooks/useTransactionDeadline'
+import { BigNumber } from 'ethers'
+import { useAddPendingOrder } from '../state/operator/hooks'
+import { AddPendingOrderParams, OrderCreation, OrderID, OrderKind, OrderStatus } from '../state/operator/actions'
+import { isAddress, shortenAddress } from '@src/utils'
+
+interface PostOrderParams {
+  account: string
+  chainId: ChainId
+  trade: Trade
+  validTo: number
+  recipient: string
+  recipientAddressOrName: string | null
+  addPendingOrder: (order: AddPendingOrderParams) => void
+}
+
+const MAX_VALID_TO_EPOCH = BigNumber.from('0xFFFFFFFF').toNumber() // Max uint32 (Feb 07 2106 07:28:15 GMT+0100)
+const DEFAULT_APP_ID = 0
+
+type UnsignedOrder = Omit<OrderCreation, 'signature'>
+
+function getSummary(params: PostOrderParams): string {
+  const { trade, account, recipient, recipientAddressOrName } = params
+
+  const inputSymbol = trade.inputAmount.currency.symbol
+  const outputSymbol = trade.outputAmount.currency.symbol
+  const inputAmount = trade.inputAmount.toSignificant(3)
+  const outputAmount = trade.outputAmount.toSignificant(3)
+
+  const base = `Swap ${inputAmount} ${inputSymbol} for ${outputAmount} ${outputSymbol}`
+  const withRecipient =
+    recipient === account
+      ? base
+      : `${base} to ${
+          recipientAddressOrName && isAddress(recipientAddressOrName)
+            ? shortenAddress(recipientAddressOrName)
+            : recipientAddressOrName
+        }`
```

Comment:
I agree! I find uniswap code sometimes hard to read with combined ternary conditions

This was copy paste, I didn't want to fix their code, but since now is our side, I'll re-write simpler

Reply:
I hope is easier now accab0dc71e7612c50d89557aa97fd65736c669e